src/store/index-level.ts

```diff
@@ -76,33 +76,36 @@ export class IndexLevel {
       throw new DwnError(DwnErrorCode.IndexMissingIndexableProperty, 'Index must include at least one valid indexable property');
     }
 
-    const indexOps: LevelWrapperBatchOperation<string>[] = [];
+    const item: IndexedItem = { messageCid, indexes };
+    const indexOpPromises: Promise<LevelWrapperBatchOperation<string>>[] = [];
```

Comment:
Can we rename: `opCreationPromises`

Reply:
Went with `putOperationPromises`

Reply:
Definitely not a battle I wanna engaged in, as it is arguable a nit, but just adding context to my thought: I thought the word "creation" is important for describing what the promises are because without it it would appear that the array is holding the result of "execution" of put operations themselves.

Reply:
Ah yeah I understand what you were trying to convey there. I renamed it and also changed the name of `generateDeleteIndexedItemOperation` to `createDeleteIndexedItemOperation` so that it fits better in the delete method.